tests/unit/daemons_test.py

```diff
@@ -65,145 +66,171 @@ class DaemonsStarterTestCase(TestCase, integration.SaltClientTestCaseMixIn):
     Unit test for the daemons starter classes.
     '''
 
+    def _multiproc_exec_test(self, exec_test):
+        m_parent, m_child = multiprocessing.Pipe()
+        p_ = multiprocessing.Process(target=exec_test, args=(m_child,))
+        p_.start()
+        self.assertTrue(m_parent.recv())
+        p_.join()
+
     def test_master_daemon_hash_type_verified(self):
         '''
         Verify if Master is verifying hash_type config option.
 
         :return:
         '''
-        def _create_master():
-            '''
-            Create master instance
-            :return:
-            '''
-            master = daemons.Master()
-            master.config = {'user': 'dummy', 'hash_type': alg}
-            for attr in ['master', 'start_log_info', 'prepare']:
-                setattr(master, attr, MagicMock())
-
-            return master
-
-        _logger = LoggerMock()
-        with patch('salt.cli.daemons.check_user', MagicMock(return_value=True)):
-            with patch('salt.cli.daemons.log', _logger):
-                for alg in ['md5', 'sha1']:
-                    _create_master().start()
-                    self.assertEqual(_logger.last_type, 'warning')
-                    self.assertTrue(_logger.last_message)
-                    self.assertTrue(_logger.last_message.find('Do not use {alg}'.format(alg=alg)) > -1)
-
-                _logger.reset()
-
-                for alg in ['sha224', 'sha256', 'sha384', 'sha512']:
-                    _create_master().start()
-                    self.assertEqual(_logger.last_type, None)
-                    self.assertFalse(_logger.last_message)
+        def exec_test(child_pipe):
+            def _create_master():
+                '''
+                Create master instance
+                :return:
+                '''
+                obj = daemons.Master()
+                obj.config = {'user': 'dummy', 'hash_type': alg}
+                for attr in ['start_log_info', 'prepare', 'shutdown']:
```

Comment:
This needs a `master` attribute here, otherwise this test will hang right here:
```
test_master_daemon_hash_type_verified (unit.daemons_test.DaemonsStarterTestCase)
[CPU:100.0%|MEM:32.2%|Z:0]  ... Process Process-3:
Traceback (most recent call last):
  File "/usr/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/testing/tests/unit/daemons_test.py", line 100, in exec_test
    _create_master().start()
  File "/testing/salt/cli/daemons.py", line 207, in start
    self.master.start()
AttributeError: 'Master' object has no attribute 'master'
```